modules/oe_whitelabel_search/oe_whitelabel_search.module

```diff
@@ -49,3 +49,43 @@ function oe_whitelabel_search_preprocess_facets_summary_item_list(array &$variab
   }
   $variables['items'] = $items;
 }
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ *
+ * Add suggestions based on the block region where the form is placed.
+ */
+function oe_whitelabel_search_theme_suggestions_form_alter(array &$suggestions, array $variables): void {
+  if (!isset($variables['element']['#form_id']) || $variables['element']['#form_id'] !== 'oe_whitelabel_search_form') {
+    return;
+  }
+
+  $suggestions[] = 'form__' . $variables['element']['#region'];
+  $suggestions[] = 'form__' . $variables['element']['#region'] . '__' . $variables['element']['#form_id'];
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ *
+ * Specify a suggestion for the form_element based on the form id.
+ */
+function oe_whitelabel_search_theme_suggestions_form_element_alter(array &$suggestions, array $variables): void {
+  if (!isset($variables['element']['#form_id']) || $variables['element']['#form_id'] !== 'oe_whitelabel_search_form') {
+    return;
+  }
+
+  $suggestions[] = 'form_element__' . $variables['element']['#form_id'];
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ *
+ * Add region suggestions for the oe_whitelabel_search_form input field.
+ */
+function oe_whitelabel_search_theme_suggestions_input_alter(array &$suggestions, array $variables): void {
+  if (!isset($variables['element']['#form_id']) || $variables['element']['#form_id'] !== 'oe_whitelabel_search_form') {
+    return;
+  }
+  $suggestions[] = $variables['theme_hook_original'] . '__' . $variables['element']['#region'];
+  $suggestions[] = $variables['theme_hook_original'] . '__' . $variables['element']['#region'] . '__' . $variables['element']['#form_id'];
+}
```

Comment:
I found a way we can avoid all these `hook_theme_suggestions_HOOK_alter()` functions.
We can put the theme hook suggestion directly into the form builder class.
See https://github.com/openeuropa/oe_whitelabel/commit/2a5695e4b6cf4c3ff132ff1cb0b0a005afec7f4e

Reply:
Besides this, I am skeptical about how we use theme hook suggestions here.
But I heard it was already agreed on with @drishu, so don't change it just for me.

I will still share my thoughts:
- I think I would put `'__' . $form_id . '__' . $region` instead of `'__' . $region . '__' . $form_id`, because this would make a more natural fallback hierarchy.
- Perhaps we should not use the region as a selector, but something else, e.g. a checkbox in the block config, or by having two different block plugins. By using the region, we make life harder for sub-themes that have their own region names. I heard that apparently I said to use the region in the refinement, in that case, my bad!
- Some changes could be done with render element altering instead of custom templates. Either in the form builder class or the block plugin class, or using `hook_block_view_BASE_BLOCK_ID_alter()`.

But as said, don't follow these suggestions unless @drishu or @brummbar agree.